llvm/lib/CodeGen/SelectionDAG/LegalizeDAG.cpp

```diff
@@ -1377,6 +1378,20 @@ void SelectionDAGLegalize::LegalizeOp(SDNode *Node) {
   }
 }
 
+// Helper function that generates an MMO that considers the alignment of the
+// stack, and the size of the stack object
+static MachineMemOperand *getStackAlignedMMO(SDValue StackPtr,
+                                             MachineFunction &MF) {
+  auto &MFI = MF.getFrameInfo();
+  int FI = cast<FrameIndexSDNode>(StackPtr)->getIndex();
+  MachinePointerInfo PtrInfo = MachinePointerInfo::getFixedStack(MF, FI);
+  MachineMemOperand *MMO =
+      MF.getMachineMemOperand(PtrInfo, MachineMemOperand::MOStore,
+                              MFI.getObjectSize(FI), MFI.getObjectAlign(FI));
```

Comment:
@davemgreen do we have some representation for Unknown size? 

Here is what I propose to change this function to:
```
static MachineMemOperand *getStackAlignedMMO(SDValue StackPtr,
                                             MachineFunction &MF, bool isScalableObject) {
  auto &MFI = MF.getFrameInfo();
  int FI = cast<FrameIndexSDNode>(StackPtr)->getIndex();
  MachinePointerInfo PtrInfo = MachinePointerInfo::getFixedStack(MF, FI);
  auto objectSize = isScalableObject ? UNKNOWN_SIZE : MFI.getObjectSize();
  MachineMemOperand *MMO =
      MF.getMachineMemOperand(PtrInfo, MachineMemOperand::MOStore,
                              objectSize, MFI.getObjectAlign(FI));
  return MMO;
}
```
but I do not know if this is feasible in the current framework. 

I also considered passing an `LLT`as argument to `MF.getMachineMemOperand` which is initialized with `LLT::scalable_vector`, however I was unsure of what to pass as arguments to this function in this case. Can you help me out?

Reply:
If this is the uint64_t accessor to getMachineMemOperand then I think you can use ~UINT64_C(0) for an unknown size. There is MemoryLocation::UnknownSize that should be the same value.

There is a patch in #70452 to clean up some of this and make scalable sizes more naturally supported. That patch is a bit sprawling though, so difficult to get past review in its current state. It doesn't alter the MFI.getObjectSize size, but changes some of the other interfaces to handle scalable TypeSizes a little better.

Reply:
@davemgreen Thanks for the suggestion! I have included your suggestions in the latest version of the patch, and there was no requirement to change the test `sve-extract-fixed-vector.ll`